From 87662455fc6eae38de70fd7b6f758c22ec7a443a Mon Sep 17 00:00:00 2001
Message-Id: <87662455fc6eae38de70fd7b6f758c22ec7a443a.1526634690.git.lucien.xin@gmail.com>
In-Reply-To: <789591c7318a0423984037b6a8ba3f3925f4eaa7.1526634690.git.lucien.xin@gmail.com>
References: <789591c7318a0423984037b6a8ba3f3925f4eaa7.1526634690.git.lucien.xin@gmail.com>
From: Timothy Redaelli <tredaelli@redhat.com>
Date: Tue, 10 Apr 2018 15:54:02 +0200
Subject: [PATCHv2 2/3] configure.ac: Empty LDFLAGS before checking for libnl3

Currently since CFLAGS are dropped if you have LDFLAGS=-pie (default on RHEL)
the rtnl_link_get_phys_port_id, rtnl_link_set_carrier and rtnl_link_get_carrier
tests always fails:

/usr/bin/ld: /tmp/ccv5GdFD.o: relocation R_X86_64_PC32 against undefined symbol
`rtnl_link_get_carrier@@libnl_3' can not be used when making a shared object;
recompile with -fPIC

This commits empty LDFLAGS before launching the 3 tests and restores it
after the tests.

Signed-off-by: Timothy Redaelli <tredaelli@redhat.com>
Signed-off-by: Jiri Pirko <jiri@mellanox.com>
---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 60657bb..f27c15c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -39,8 +39,10 @@ LT_INIT
 
 PKG_CHECK_MODULES([LIBNL], [libnl-3.0 libnl-genl-3.0 libnl-route-3.0 libnl-cli-3.0])
 	TMP_CFLAGS="$CFLAGS"
+	TMP_LDFLAGS="$LDFLAGS"
 	TMP_LIBS="$LIBS"
 	CFLAGS="$CPPFLAGS $LIBNL_CFLAGS"
+	LDFLAGS=""
 	LIBS="$LIBS $LIBNL_LIBS"
 	AC_CHECK_LIB([nl-route-3], [rtnl_link_get_phys_port_id],
 		     AC_DEFINE(HAVE_RTNL_LINK_GET_PHYS_ID, [1], [Define to 1 if you have rtnl_link_get_phys_port_id function.]))
@@ -49,6 +51,7 @@ PKG_CHECK_MODULES([LIBNL], [libnl-3.0 libnl-genl-3.0 libnl-route-3.0 libnl-cli-3
 	AC_CHECK_LIB([nl-route-3], [rtnl_link_get_carrier],
 		     AC_DEFINE(HAVE_RTNL_LINK_GET_CARRIER, [1], [Define to 1 if you have rtnl_link_get_carrier.]))
 	CFLAGS="$TMP_CFLAGS"
+	LDFLAGS="$TMP_LDFLAGS"
 	LIBS="$TMP_LIBS"
 
 PKG_CHECK_MODULES([LIBDAEMON], [libdaemon])
-- 
2.1.0

